### PR TITLE
adding code support to deploy Custom CRUSH rule EC pools

### PR DESCRIPTION
**Pull request by pdhiran**
Adding custom crush rules to deploy EC 8+6 pool.

### CODE REVIEW

File: ceph/rados/crushtool_workflows.py [hunk 1]
```diff
@@ -640,3 +640,25 @@ def test_crush_map_bin(
             return False, " "
         log.debug(f"Successfully tested the bin file for {test} and output : {out}")
         return True, out
+
+    def add_crush_rule(self, rule_name: str, rules: str):
```
Comment (harshkumarRH) [inline ceph/rados/crushtool_workflows.py:644]: Very minor - would be good to include the return type in doc string

File: ceph/rados/crushtool_workflows.py [hunk 2]
```diff
@@ -640,3 +640,25 @@ def test_crush_map_bin(
...
+        # Adding the crush rules into the file
+        rule_str = f"rule {rule_name} {{ \n{rules}\n}}"
+        cmd = f"echo '{rule_str}' >> {loc}"
+        self.client.exec_command(cmd=cmd, sudo=True)
+
+        _, loc = self.compile_crush_map_txt(source_loc=loc)
+        if not self.set_crush_map_bin(loc=loc):
+            log.error(f"Failed to set rules {rules} on the cluster")
+            return False
+        log.info(f"Successfully set rules {rules} with name {rule_name} on the cluster")
```
Comment (harshkumarRH) [inline ceph/rados/crushtool_workflows.py:663]: Just a hunch, but wouldn't this statement become very huge and disoriented if we include the `{rules}` here? Maybe just mentioning the rule_name is enough. WDYT?

Comment (pdhiran) [reply ceph/rados/crushtool_workflows.py:663]: Yes, Sounds good. Removed the rules string.

**Review by harshkumarRH — DISMISSED** [review]
LGTM, requested changes are yet to be pushed

File: tests/rados/test_ec_86.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    crush_rule = config.get("crush_rule", "rule-86")
+    crush_rules = {
+        "rule-86": """id 86
+type erasure
+step set_chooseleaf_tries 5
+step set_choose_tries 100
+step take default
+step choose indep 4 type host
+step chooseleaf indep 4 type osd
```
Comment (viduship) [inline tests/rados/test_ec_86.py:37]: ensure we have choose instead of chooseleaf

Comment (pdhiran) [reply tests/rados/test_ec_86.py]: Updated!

**Review by harshkumarRH — APPROVED** [review]
LGTM

**Review by neha-gangadhar — APPROVED** [review]
Approving based on review completed and the comments addressed.

**Pull request merged**